### DISCUSSION TITLE
C# 컨벤션관련 글

### DISCUSSION THREAD

**Discussion by fkdl0048** [The-Day-In-1984/1984_Project#98] [General]
C# 표준 컨벤션을 따라가자고 했는데 표기법이나 정확한 명칭이 없어 혼동이 있는 것 같아서 잘 정리된 글을 참고.  

https://everyday-devup.tistory.com/21

**Comment by ccmuk58** [comment]
아 맞다 회의 때 말하려다가 깜빡했는데, 
보통 static 선언 할때 `MAX_LEVEL` 이런 형태로 많이들 알고 있는데, 이건 JAVA에서 주로 사용하는 컨벤션이더라고
물론 C#에서도 많이 쓰긴 하는 것 같은데, 
C# 공식 문서를 보면 따로 정의된 사항은 없지만 종종 표기되어 있는 상수들을 보면 `MaxLevel`과 같은 형태로 파스칼 표기법으로 되어있어.
그래서 나도 일단 그렇게 작성하긴 했는데 이렇게 하면 될 듯..?

다시 확인 해보니까 private은 `s_`접두사 붙이고 카멜표기법으로 하더라고

요약
```c#
private static int s_maxLevel = 0;
public static int MaxLevel = 0;
```
public 및 private static 은 위와 같이 작성하도록 하는게 어떨까. (사실 나도 이미 틀림)

https://learn.microsoft.com/ko-kr/dotnet/csharp/fundamentals/coding-style/coding-conventions
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

**Reply by fkdl0048** [reply]
오오... 좋아 C#자체가 자바랑 C++합친 언어라 되게 헷갈린다고 하더라고..

**Comment by kyoonge** [comment]
헐 정리천재들... 감사합니다
